opentracing-observer/src/main/java/io/opentracing/contrib/observer/SpanData.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.observer;
+
+import java.util.Map;
+
+/**
+ * This interface provides information about the current {@link io.opentracing.Span} to the observer
+ * methods.
+ *
+ */
+public interface SpanData {
+
+    /**
+     * This method returns an id that can be used for correlate actions invoked on a
+     * stateful observer. It should only be used within the scope of an application,
+     * to uniquely distinguish one span from another, to enable state to be maintained
+     * within observer implementation where appropriate.
+     *
+     * @return The correlation id for the span, MUST implement equals/hashCode to enable
+     *              it to be used as a map key
+     */
+    Object getCorrelationId();
```

Comment:
Maybe create a separate issue for that? Or should we agree on something before merging? Also, shouldn't the default equals/hashCode implementation (object identity) be good enough?